src/sparsematrix.jl

```diff
@@ -1632,6 +1697,22 @@ function fkeep!(A::AbstractSparseMatrixCSC, f, trim::Bool = true)
     return A
 end
 
+function _fkeep!_fixed(A::AbstractSparseMatrixCSC, f) where S
+    @inbounds for j in axes(A, 2)
+        for k in getcolptr(A)[j]:getcolptr(A)[j+1]-1
+            i = rowvals(A)[k]
+            x = nonzeros(A)[k]
+            # If this element should be kept, rewrite in new position
+            if f(Ai, Aj, Ax)
+                nonzeros(A)[k] = zero(eltype(A))
+            end
+        end
+    end
+    return A
+end
+
+fkeep!(A::AbstractSparseMatrixCSC, f) = _is_fixed(A) ? _fkeep!_fixed(A, f) : _fkeep!(A, f)
```

Comment:
after this patch is done, i'll fix fkeep! once and for all. the function is always first.